### DISCUSSION TITLE
.ook huge file size!

### DISCUSSION THREAD

**Discussion by frevoli** [merbanan/rtl_433#2399] [Q&A]
Hello community.
I save the output of my rtl_433 session to file using the “-W ook” option.
The problem is that after a relatively short time this file is huge! For example, after just 12 hours it is +2Gb! 
I can store and work with this amount of data if it has to be this size, but it really needs to be split over several files. As far as I can tell, only one file is created. 
does anyone have any advice as to how I could either get rtl_433 to start a new file every ‘X’ Mb, or make the file size smaller in the first place?
Many thanks in advance!
Matt

**Comment by zuckschwerdt** [comment]
That (`-W ook`) won't work. You need to have a file extension (`.ook`) or override (`OOK:`).
I.e. use `-w myfile.ook` or use `-w OOK:myfile` (the later is not recommended).

Do you really want to log all received transmissions as pulse text (ook)?
Maybe we should add an option to the "Signal auto save" (`-S`) to save pulse text instead of I/Q data, then you get one file per transmission.

**Reply by frevoli** [reply]
Sorry, my mistake. I do use -W filename.ook

**Reply by frevoli** [reply]
I am logging all signals over a given time, to post process after. So I do need to be able to record everything above a certain SNR. If the size has to be that large then that's ok, but a way of creating a new ook file when it reaches a certain size would be great.  : )

**Reply by frevoli** [reply]
One file per transmission would also work. Perfect

**Reply by zuckschwerdt** [reply]
For now you can split yourself. Pipe the data, e.g. `rtl_433 -w OOK:- |splitter.py` and write the Python script to split on `;received` (the first line in every chunk) once a set size is reached or a given time passed.

**Reply by frevoli** [reply]
Ok that's great. I'll do exactly that. Thank you!
